lib/honeybadger/client.ex

```diff
@@ -71,11 +72,32 @@ defmodule Honeybadger.Client do
   # Callbacks
 
   def init(state) do
+    warn_if_incomplete_env(state)
+    warn_in_dev_mode(state)
     :ok = :hackney_pool.start_pool(__MODULE__, [max_connections: @max_connections])
 
     {:ok, state}
   end
 
+  @mandatory_keys ~w[api_key environment_name]a
+  defp warn_if_incomplete_env(%{enabled: true}) do
+    @mandatory_keys
+    |> Enum.each(fn key ->
+      if !Honeybadger.get_env(key) do
+        Logger.error("mandatory :honeybadger config key #{key} not set")
```

Comment:
I think this should be `Logger.warn`, like the dev mode warning. This isn't really an error.

Reply:
This runs only in a non excluded env. Shouldn't we print an error if `api_key` is missing? This however checks for both `api_key` and `environment_name`

Reply:
Ah yeah, that sounds right.